TAScheduler/UnitTests/tests.py

```diff
@@ -178,50 +189,778 @@ class TestCourseGetCrseInfo(TestCase):  # Randall
+            section_id=1011,
+            course=tmpcourse,
+            location="Cool place",
+            meeting_time=datetime.datetime
+        )
+
+        tmplec = Lecture.objects.create(
+            section=tmpsection,
+            ta=tmpta,
+        )
+
+        tempuser2 = User(
+            email_address="test@instructor.com",
+            password="password",
+            first_name="firstin",
+            last_name="lastin",
+            home_address="Your mom's house",
+            phone_number=1234567890
+        )
+
+        self.instructor = Instructor.objects.create(
+            user=tempuser2
+        )
+        self.instructor.save()
+
+        self.lecture = LectureObj(tmplec)
+
+    def test_get_instructor(self):
+        self.lecture.addInstr(self.instructor)
+        self.assertEquals(self.instructor, self.lecture.getLecInstrAsmgt(),
+                          "getLecInstrAsmgt() does not return correct")
+
+    def test_get_with_no_instructor(self):
+        self.assertIsNone(self.lecture.getLecInstrAsmgt(), "getLecInstrAsmgt() Retrieved instructor when none exists")
+
+
+class TestLectureAddInstructor(unittest.TestCase):  # Joe
+    lecture = None
+    instructor = None
+
+    def setUp(self):
+        tempuser = User(
+            email_address="test@ta.com",
+            password="password",
+            first_name="first",
+            last_name="last",
+            home_address="Your mom's house",
+            phone_number=1234567890
+        )
+
+        tmpta = TA.objects.create(
+            user=tempuser,
+            grader_status=True
+        )
+
+        tempuser2 = User(
+            email_address="test@instructor.com",
+            password="password",
+            first_name="firstin",
+            last_name="lastin",
+            home_address="Your mom's house",
+            phone_number=1234567890
+        )
+
+        self.instructor = Instructor.objects.create(
+            user=tempuser2
+        )
+        self.instructor.save()
+
+        tmpcourse = Course.objects.create(
+            course_id=100,
+            semester="fall 2023",
+            name="testCourse",
+            description="test",
+            num_of_sections=3,
+            modality="online",
+            credits=3
+        )
+
+        tmpsection = Section.objects.create(
+            section_id=1011,
+            course=tmpcourse,
+            location="Cool place",
+            meeting_time=datetime.datetime
+        )
+
+        tmplec = Lecture.objects.create(
+            section=tmpsection,
+            ta=tmpta,
+        )
+
+        self.info = {
+            "ta", tmpta,
+            "section", tmpsection,
+        }
+
+        self.lecture = LectureObj(tmplec)
+
+    def test_add_but_full(self):
+        self.lecture.addInstr(self.instructor)
+        with self.assertRaises(RuntimeError, msg="Tried to add Instructor to full lecture"):
+            self.lecture.addInstr(self.instructor)
```

Comment:
same^, lecture already has a user attached to it so adding another user (line 673) wouldn't work